framework/src/meshgenerators/SmoothMeshGenerator.C

```diff
@@ -109,27 +109,24 @@ SmoothMeshGenerator::SmoothMeshGenerator(const InputParameters & parameters)
 std::unique_ptr<MeshBase>
 SmoothMeshGenerator::generate()
 {
-  std::unique_ptr<MeshBase> mesh = nullptr;
+  std::unique_ptr<MeshBase> input = std::move(_input);
+  std::unique_ptr<UnstructuredMesh> mesh = dynamic_pointer_cast<UnstructuredMesh>(input);
   std::unique_ptr<libMesh::MeshSmoother> smoother = nullptr;
 
   if (_algorithm == "laplace")
   {
-    std::unique_ptr<MeshBase> old_mesh = std::move(_input);
-    if (!old_mesh->is_replicated())
+    if (!mesh->is_replicated())
```

Comment:
Now that I think about it - could we weaken this to `if (!mesh->is_serial())`?  That's the actual requirement that we need for the libMesh Laplace smoother - we don't care if the mesh could be distributed later, just as long as it's serialized at the moment.